cairo/tests/ethereum/utils/test_numeric.py

```diff
@@ -240,70 +135,210 @@ def test_Bytes32_from_be_bytes(self, cairo_run, bytes: Bytes):
+
+        @given(a=..., b=...)
+        def test_U256_max(self, cairo_run, a: U256, b: U256):
+            result = cairo_run("U256_max", a, b)
+            expected = max(a, b)
+            assert result == expected
+
+    class TestU384:
+        @given(value=...)
+        def test_U384_is_one(self, cairo_run, value: U384):
+            cairo_result = cairo_run("U384_is_one", value)
+            assert cairo_result == 1 if value == U384(1) else cairo_result == 0
+
+        @given(bytes=st.binary(max_size=512))
+        def test_U384_from_be_bytes(self, cairo_run, bytes: Bytes):
+            try:
+                result = cairo_run("U384_from_be_bytes", bytes)
+            except ValueError:
+                assert len(bytes) > 48
+                return
+
+            expected = U384(int.from_bytes(bytes, "big"))
+
+            assert result == expected
+
+        @given(bytes=st.binary(max_size=512))
+        def test_U384_from_le_bytes(self, cairo_run, bytes: Bytes):
+            try:
+                result = cairo_run("U384_from_le_bytes", bytes)
+            except ValueError:
+                assert len(bytes) > 48
+                return
+
+            expected = U384(int.from_bytes(bytes, "little"))
+            assert result == expected
+
+        @given(a=..., b=...)
+        def test_U384__eq__(self, cairo_run, a: U384, b: U384):
+            assert (a == b) == cairo_run("U384__eq__", a, b)
+
+        @given(value=...)
+        def test_U384_is_zero(self, cairo_run, value: U384):
+            cairo_result = cairo_run("U384_is_zero", value)
+            assert cairo_result == 1 if value == U384(0) else cairo_result == 0
+
+        @given(value=...)
+        def test_get_u384_bits_little(self, cairo_run, value: U384):
+            (cairo_bits_ptr, cairo_bits_len) = cairo_run("get_u384_bits_little", value)
+
+            python_bits = get_u384_bits_little(value)
+            cairo_bits = [cairo_bits_ptr[i] for i in range(cairo_bits_len)]
+            assert python_bits == cairo_bits, f"Failed for value {value}"
+
+        @given(value=...)
+        def test_U384_to_be_bytes(self, cairo_run, value: U384):
+            int_value = value._number
+            if int_value == 0:
+                length = 1  # At least one byte for zero
+            else:
+                length = (int_value.bit_length() + 7) // 8
+
+            cairo_result = cairo_run("U384_to_be_bytes", value, length)
+
+            expected_bytes = int_value.to_bytes(length, "big")
+            assert len(cairo_result) == length
+            assert bytes(cairo_result) == expected_bytes
+
+        @given(value=...)
+        def test_U384_to_le_bytes(self, cairo_run, value: U384):
+            int_value = value._number
+            if int_value == 0:
+                length = 1  # At least one byte for zero
+            else:
+                length = (int_value.bit_length() + 7) // 8
+
+            cairo_result = cairo_run("U384_to_le_bytes", value, length)
+
```

Comment:
In the test, length should be a parameter, not always strictly equal to the number of bytes required to read the word (as in the cairo function, length is an arbitrary number). What should be the behavior is length < number of bytes in the word?

Reply:
you are raising a good point. In the context of U384_to_le_bytes, it is always used for a length of 48bytes. So I think it should be refactored: change the name `U384_to_le_48_bytes` and remove the length parameter.

Reply:
Ok for now but generally I think we should make our type libraries generic